doc/tutorials/enable_s5.rst

```diff
@@ -0,0 +1,367 @@
+
+ACRN provides a mechanism to trigger the S5 state transition throughout the system.
+It uses vUART channel to communicate between the Service VM and the User VMs.
+The diagram below shows the overall architecture:
+
+.. figure:: images/s5_overall_architecture.png
+   :align: center
+
+   S5 overall architecture
+
+- **Scenario I**:
+
+    The User VM's serial port device (``ttySn``) is emulated in Device Model, the channel from Service VM to User VM:
+
+    .. graphviz:: images/s5-scenario-1.dot
+       :name: s5-scenario-1
+
+- **Scenario II**:
+
+    The User VM's (like RT-Linux or other RT-VMs) serial port device (``ttySn``) is emulated in Hypervisor,
+    the channel from Service OS to User VM:
+
+    .. graphviz:: images/s5-scenario-2.dot
+       :name: s5-scenario-2
+
+How to trigger User VM's S5?
+============================
+
+On Service VM side, it uses the ``acrnctl`` tool to trigger the User VM's S5 flow:
+``acrnctl stop user-vm-name``. Then Device Model will send a ``shutdown`` command
+to User VM through channel, if User VM receives the command, it will send an "ACK"
+to Device Model. It is the responsibility of Service VM to check if the User VMs
+shutdown successfully or not, and decides when to power off itself.
+
+What does "life cycle manager" in User VM do?
+=============================================
+
+As current S5 reference design, there is a life-cycle-manager daemon running in
+User VM to implement S5. It will wait for the command from Service VM on the
+paired serial port, there is a simple protocol between Service VM and User VM:
+when it receives ``shutdown``, then send "acked" to Service VM, then it can power off
+User VM.
+
+However, if the User VM is not ready to power off, it can ignore the ``shutdown`` command.
+
+.. _enable_s5:
+
+Enable S5
+*********
+
+Enable S5 based on the the particular OS:
+
+* For Linux (LaaG) or Windows (WaaG), refer to the following configurations in
+  ``devicemodel/samples/nuc/launch_uos.sh`` launch script for ``acrn-dm``.
+
+  .. code-block:: none
+     :emphasize-lines: 2,3,4,17
+     :name: laag-waag-script
+     :caption: LaaG/WaaG launch script
+  
+     #for pm by vuart setting
+     pm_channel="--pm_notify_channel uart "
+     pm_by_vuart="--pm_by_vuart pty,/run/acrn/life_mngr_"$vm_name
+     pm_vuart_node=" -s 1:0,lpc -l com2,/run/acrn/life_mngr_"$vm_name
+     
+     #for memsize setting
+     mem_size=2048M
+     
+     acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+       -s 2,pci-gvt -G "$2" \
+       -s 5,virtio-console,@stdio:stdio_port \
+       -s 6,virtio-hyper_dmabuf \
+       -s 3,virtio-blk,/home/clear/uos/uos.img \
+       -s 4,virtio-net,tap0 \
+       -s 7,virtio-rnd \
+       --ovmf /usr/share/acrn/bios/OVMF.fd \
+       $pm_channel $pm_by_vuart $pm_vuart_node \
+       $logger_setting \
+       --mac_seed $mac_seed \
+       $vm_name
+     }
+
+* For RT-Linux, refer to the ``devicemodel/samples/nuc/launch_hard_rt_vm.sh`` script:
+
+  .. code-block:: none
```

Comment:
Same concern here with copying some lines from a script vs. including them directly from the source file.